therapy/etl/merge.py

```diff
@@ -0,0 +1,188 @@
+"""Create concept groups and merged records."""
+from therapy.schemas import SourceName, NamespacePrefix
+from therapy.database import Database
+from typing import Set, Dict
+import logging
+from timeit import default_timer as timer
+
+logger = logging.getLogger('therapy')
+logger.setLevel(logging.DEBUG)
+
+DISALLOWED_SOURCES = {NamespacePrefix.DRUGBANK.value,
+                      NamespacePrefix.CHEMBL.value}
+
+
+class Merge:
+    """Handles record merging."""
+
+    def __init__(self, database: Database):
+        """Initialize Merge instance.
+
+        :param Database database: db instance to use for record retrieval
+            and creation.
+        """
+        self._database = database
+        self._groups = {}  # dict keying concept IDs to group Sets
+
+    def create_merged_concepts(self, record_ids: Set[str]):
+        """Create concept groups, generate merged concept records, and
+        update database.
+
+        :param Set[str] record_ids: concept identifiers from which groups
+            should be generated. Should *not* include any records from
+            excluded sources.
+        """
+        logger.info('Generating record ID sets...')
+        start = timer()
+        for record_id in record_ids:
+            new_group = self._create_record_id_set(record_id)
+            if new_group:
+                for concept_id in new_group:
+                    self._groups[concept_id] = new_group
+        end = timer()
+        logger.debug(f'Built record ID sets in {end - start} seconds')
+
+        logger.info('Creating merged records and updating database...')
+        uploaded_ids = set()
+        start = timer()
+        for record_id, group in self._groups.items():
+            if record_id in uploaded_ids:
+                continue
+            merged_record = self._generate_merged_record(group)  # noqa
```

Comment:
Can we remove this? I'm unsure of the error it's catching? 
```suggestion
            merged_record = self._generate_merged_record(group)
```